doc/source/release-notes/Percona-XtraDB-Cluster-5.7.19-29.22.rst

```diff
@@ -0,0 +1,119 @@
+.. rn:: 5.7.19-29.22
```

Comment:
We need to capture essence of this email somewhere.

<mail>
Recent PXC release addresses one of the major bug related to difference in binlog handling.
(https://bugs.launchpad.net/percona-xtradb-cluster/+bug/1690398)

It also exposes one major gap in SST process.
JOINER and DONOR doesn't communicate their MySQL version to each other making it difficult to take version specific decision.

As part of fix for this bug we also introduced a version communication as part of SST protocol.
It is also needed to handle the extra information that DONOR will send to JOINER needed for the said bug.

---------------------

So far so good. How does it affect the general PXC ?

* SST protocol change is part of PXC-5.6.37 and PXC-5.7.19.

* This means older version of PXC can act as DONOR and new version will still understand old protocol (backward compatible).

* But older version of PXC can't act as JOINER with new version as DONOR. (Protocol can't be forward compatible w/o versioning).

Needless to mention introducing version handshaking make it future proof and flexible in handling version specific request.
</mail>

Reply:
This is captured in descriptions for PXC-827 and PXC-851